src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -38,31 +38,34 @@ function ShallowWaterSolutionFields(arch, grid, bcs)
     return (uh=uh, vh=vh, h=h)
 end
 
-struct ShallowWaterModel{G, A<:AbstractArchitecture, T, V, R, F, E, Q, C, K, TS} <: AbstractModel{TS}
+struct ShallowWaterModel{G, A<:AbstractArchitecture, T, V, R, F, E, B, Q, C, K, TS} <: AbstractModel{TS}
     
-                 grid :: G         # Grid of physical points on which `Model` is solved
-         architecture :: A         # Computer `Architecture` on which `Model` is run
-                clock :: Clock{T}  # Tracks iteration number and simulation time of `Model`
-            advection :: V         # Advection scheme for velocities _and_ tracers
-             coriolis :: R         # Set of parameters for the background rotation rate of `Model`
-              forcing :: F         # Container for forcing functions defined by the user
-              closure :: E         # Diffusive 'turbulence closure' for all model fields
-             solution :: Q         # Container for transports `uh`, `vh`, and height `h`
-              tracers :: C         # Container for tracer fields
-        diffusivities :: K         # Container for turbulent diffusivities
-          timestepper :: TS        # Object containing timestepper fields and parameters
+                          grid :: G         # Grid of physical points on which `Model` is solved
+                  architecture :: A         # Computer `Architecture` on which `Model` is run
+                         clock :: Clock{T}  # Tracks iteration number and simulation time of `Model`
+    gravitational_acceleration :: T         # Gravitational acceleration, full, or reduced
+                     advection :: V         # Advection scheme for velocities _and_ tracers
+                      coriolis :: R         # Set of parameters for the background rotation rate of `Model`
+                       forcing :: F         # Container for forcing functions defined by the user
+                       closure :: E         # Diffusive 'turbulence closure' for all model fields
+                    bathymetry :: B         # Diffusive 'turbulence closure' for all model fields
+                      solution :: Q         # Container for transports `uh`, `vh`, and height `h`
+                       tracers :: C         # Container for tracer fields
+                 diffusivities :: K         # Container for turbulent diffusivities
+                   timestepper :: TS        # Object containing timestepper fields and parameters
 
 end
 
 function ShallowWaterModel(;
                            grid,
+                           gravitational_acceleration,
   architecture::AbstractArchitecture = CPU(),
-                          float_type = Float64,
-                               clock = Clock{float_type}(0, 0, 1),
+                               clock = Clock{eltype(grid)}(0, 0, 1),
```

Comment:
I don't know why we have this and am happy with whatever is agreed upon

Reply:
@ali-ramadhan the problem is that `float_type` was never used. I think we should use `grid` to coordinate `float_type` rather than having it as an argument in the model constructors. What do you think?

Reply:
I agree that we're not really using the `float_type` kwarg so might be good to remove it.

I'm hoping to add some tests/utils to ensure type stability for models, i.e. constructing a model with `Posit16` ensures that every number in the model is `Posit16`, etc. But for this the `float_type` kwarg may not be needed as it's available via `eltype(grid)`.

Reply:
Just want to be clear, it's not that we "weren't really" using `float_type` (as users); it was literally not used in the constructor except in `Clock` and thus did not seem to be serving its apparent purpose.

Reply:
Note that `eltype(grid)` is used where it matters in constructing field tuples

Reply:
True, it's a remnant from very early versions of Oceananigans.

Reply:
I presume we are happy with what is proposed so will resolve the conversation.